### PR TITLE
fix(csv-outputs): compliance outputs not showing consistents values

### PR DESCRIPTION
**Pull request by pedrooot**
### Description

Compliance outputs were showing wrong values for manual checks on the field CheckID.

Screenshot 2024-06-09 at 19 03 51

In this case, for the Req 1.1.1 the checkid shown on the csv output was: "scheduler_profiling"
### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/lib/check/compliance.py [hunk 1]
```diff
@@ -70,7 +71,6 @@ def update_checks_metadata_with_compliance(
                     "Recommendation": {"Text": "", "Url": ""},
                 },
                 "Categories": [],
-                "Tags": {},
```
Comment (jfagoagas) [inline prowler/lib/check/compliance.py:73]: Why this is no longer needed?

File: prowler/compliance/kubernetes/cis_1.8_kubernetes.json
Comment (jfagoagas) [inline prowler/compliance/kubernetes/cis_1.8_kubernetes.json:1]: Why this?

File: prowler/lib/check/compliance.py [hunk 2]
```diff
@@ -34,6 +34,7 @@ def update_checks_metadata_with_compliance(
             # Save it into the check's metadata
             bulk_checks_metadata[check].Compliance = check_compliance
 
+        check_compliance = []
```
Comment (jfagoagas) [inline prowler/lib/check/compliance.py:37]: Is this the line that fixes the bug?

**Review by jfagoagas — CHANGES_REQUESTED** [review]
Please add some tests to prevent this from happening again.

Do we need to backport this to v3?

File: prowler/lib/check/compliance.py (hunk 1 above)
Comment (pedrooot) [reply prowler/lib/check/compliance.py:73]: Because it wasn't inside the Check_Metadata_Model, I've added it

File: prowler/compliance/kubernetes/cis_1.8_kubernetes.json
Comment (pedrooot) [reply prowler/compliance/kubernetes/cis_1.8_kubernetes.json:1]: To fix typos

File: prowler/lib/check/compliance.py (hunk 2 above)
Comment (pedrooot) [reply prowler/lib/check/compliance.py:37]: yes

File: prowler/lib/check/compliance.py (hunk 1 above)
Comment (jfagoagas) [reply prowler/lib/check/compliance.py:73]: add or remove?

Comment (pedrooot) [reply prowler/lib/check/compliance.py:73]: now added 😄

**Comment by pedrooot** [comment]
@jfagoagas backport to v3 added and tests

File: prowler/lib/check/compliance.py [hunk 3]
```diff
@@ -70,10 +71,10 @@ def update_checks_metadata_with_compliance(
                     "Recommendation": {"Text": "", "Url": ""},
                 },
                 "Categories": [],
-                "Tags": {},
                 "DependsOn": [],
                 "RelatedTo": [],
                 "Notes": "",
+                "Tags": {},
```
Comment (jfagoagas) [inline prowler/lib/check/compliance.py]: Why is this here? The check metadata doesn't have a `Tags` key.

File: prowler/lib/check/models.py [hunk 4]
```diff
@@ -54,6 +54,7 @@ class Check_Metadata_Model(BaseModel):
     DependsOn: list[str]
     RelatedTo: list[str]
     Notes: str
+    Tags: dict = {}
```
Comment (jfagoagas) [inline prowler/lib/check/models.py]: Why this? Sorry for the noise, in the previous comment I thought that this was being modified in another object.

File: tests/lib/check/compliance_check_test.py [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+    CIS_Requirement_Attribute,
+    CIS_Requirement_Attribute_AssessmentStatus,
+    CIS_Requirement_Attribute_Profile,
+    Compliance_Base_Model,
+    Compliance_Requirement,
+)
+from prowler.lib.check.models import Check_Metadata_Model
+
+
+class TestUpdateChecksMetadataWithCompliance:
```
Comment (jfagoagas) [inline tests/lib/check/compliance_check_test.py]:
```suggestion
class TestCompliance:
```

File: tests/lib/check/compliance_check_test.py [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+        assert len(updated_metadata["check2"].Compliance) == 1
+        assert len(updated_metadata["manual_check"].Compliance) == 1
+
+        manual_compliance = updated_metadata["manual_check"].Compliance[0]
+        assert manual_compliance.Framework == "Framework1"
+        assert manual_compliance.Provider == "Provider1"
+        assert manual_compliance.Version == "1.0"
+        assert manual_compliance.Description == "Framework 1 Description"
+
+    def test_update_checks_metadata_with_manual_control(self):
```
Comment (jfagoagas) [inline tests/lib/check/compliance_check_test.py]: What is the difference between this test and the one above? Could we assert more fields to be sure that everything is right?

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: prowler/lib/check/compliance.py (hunk 3 above)
Comment (pedrooot) [reply prowler/lib/check/compliance.py]: True!

File: prowler/lib/check/models.py (hunk 4 above)
Comment (pedrooot) [reply prowler/lib/check/models.py]: True

File: tests/lib/check/compliance_check_test.py (hunk 5 above)
Comment (pedrooot) [reply tests/lib/check/compliance_check_test.py]: Done

File: tests/lib/check/compliance_check_test.py (hunk 6 above)
Comment (pedrooot) [reply tests/lib/check/compliance_check_test.py]: Yes, one test it's enough for this behaviour

**Review by jfagoagas — APPROVED** [review]
💯
